### ISSUE TITLE
Not all possibilities are checked

### ISSUE THREAD

**Issue by osheratias** [butor/blackbird#392]
Hi,
Im running the bot and i get the output below,
im trying to understand why there is no Bitfinex/Cexio  there?
if there was, would it be a position the bot will enter(~+2.59)?

Bitfinex: 	7,359.90 / 7,360.00
Kraken: 	7,359.60 / 7,364.00
Cexio: 	7,535.10 / 7,555.70
   ----------------------------
   Bitfinex/Kraken:	-0.01% [target  0.80%, min -0.41%, max  0.56%]
   Cexio/Bitfinex:	-2.59% [target  0.80%, min -3.31%, max -2.32%]
   Cexio/Kraken:	-2.60% [target  0.80%, min -3.06%, max -2.40%]

**Comment by osheratias** [comment]
I think the problem is at main.cpp:
  params.addExchange("Cexio", params.cexioFees, false, true);

cex.io short is implemented so it shoud be 
  params.addExchange("Cexio", params.cexioFees, true, true);

**Comment by LasseMoench** [comment]
Is it actually implemented though? Both sendShortOrder and sendLongOrder seem to just call sendOrder.

**Comment by osheratias** [comment]
i think it's implemented tought im not sure :\
They both call the same function but it seems to handle both cases(like in bitfinex).

**Comment by LasseMoench** [comment]
I don't think so. First of all, a quick disclaimer. I don't have any knowledge of short selling and all that finance related stuff. Now, if I understand correctly, it seems to me that the to do short selling, you would need to use the open_position method, but the code right now seems to call place_order, and I don't think that can do short selling. However, if you believe otherwise, why don't you replace the boolean and find out what it does with a small amount of money. As I said, my understanding of the finance part isn't deep enough to conclusively say anything about this. Why don't we ask the person that coded the cexio stuff? I posted a comment on his pull request, let's hope he reads it and comes here.

**Comment by cricc78** [comment]
I couldn't fully test short selling with someone else's Cex.io account that I was using for development and I don't have my own account with them.

@osheratias is correct in changing that line of code in main.cpp. This will enable short selling for Cex.io (even if it ends up not working properly).

@LasseMoench You may be correct about needing to use open_position instead of place_order to initiate a short...the best I can recommend is change the above referenced line in main.cpp, test short selling with a small cash amount, and report back here with the results. 

If we end up needing to implement open_position for shorts, I can work on it.

**Comment by LasseMoench** [comment]
In addition to trying it out, we might also ask cex.io support how to do a short sell. I don't have an account so I can't try it, I also don't have any bitcoin funds to quickly fund a new account since I am a poorish student :) @osheratias can you just enable it and see what happens with a small amount of money? The only thing that **should** happen if it fails is that the long order completes while the short sell fails and you have to manually revert the long order.

**Comment by LasseMoench** [comment]
I created a cex.io account and funded it, gonna try it out now. However, I am 98% sure that we need open_position, since they talk about positions wherever leverage/shortselling is involved in the gui. @cricc78 would you be willing to get into it? It would be really great for everyone to have another short selling exchange apart from kraken and bitfinex, since bitfinex can't be used by people with under 10k$ anymore.

**Comment by cricc78** [comment]
Yes, I'll work on it. Please let me know your results as well.

**Comment by LasseMoench** [comment]
I will. I'm having trouble activating my api key though, and the support only sends me automatic replies on how to recover my account :(

**Comment by cricc78** [comment]
OK, I just created an account with cex.io as well, now that they are once again accepting new registrations. It won't let me activate margin trading so that I can short sell, however. Probably because I'm based in the US. Were you able to activate margin?

**Comment by LasseMoench** [comment]
Yes, I could activate it, I'm in Germany. As soon as I manage to activate my API key I'll post here, I can test everything then. Maybe I can even create an API key for you so you can test on my account.

**Comment by fdoumet** [comment]
I got in touch with the CEX.io team concerning the API key. Apparently you need to deactivate two-factor authentication, enable the API, then reenable two-factor authentication. It worked for me.

**Comment by osheratias** [comment]
@LasseMoench I will give it a go.

**Comment by monossido** [comment]
any news about that? does anyone have a working code to post a short selling on cex?
